### PR TITLE
Ori/mgmt 1220 bm-inventory should invoke the free-ip tool on the agent and validate the VIPs provided by the user

### PR DESCRIPTION
**Pull request by ori-amizur**

### CODE REVIEW

File: internal/host/freeaddressescmd_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		fCmd = NewFreeAddressesCmd(getTestLog(), "quay.io/ocpmetal/free_addresses:latest")
+
+		id = strfmt.UUID(uuid.New().String())
+		clusterId = strfmt.UUID(uuid.New().String())
+		host = getTestHost(id, clusterId, HostStatusInsufficient)
+		host.Inventory = defaultInventory()
+		Expect(db.Create(&host).Error).ShouldNot(HaveOccurred())
+	})
+
+	It("get_step", func() {
```
Comment (filanov) [inline internal/host/freeaddressescmd_test.go:34]: what about bad flows? empty up addresses?

Comment (ori-amizur) [reply internal/host/freeaddressescmd_test.go:34]: OK.  Will add.

**Comment by ori-amizur** [comment]
> Where is the part of the validation for the VIPs provided by the user?

In inventory.go

File: internal/host/freeaddressescmd.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func (f *freeAddressesCmd) prepareParam(host *models.Host) (string, error) {
+	var inventory models.Inventory
+	err := json.Unmarshal([]byte(host.Inventory), &inventory)
+	if err != nil {
+		f.log.WithError(err).Warn("Inventory parse")
+		return "", err
+	}
+	m := make(map[string]string)
```
Comment (ronniel1) [inline internal/host/freeaddressescmd.go:33]: I assume that you need a set here, and not really a map.
Better make it map[string]bool, or even map[string]struct{}

Comment (ori-amizur) [reply internal/host/freeaddressescmd.go:33]: OK.

**Pull request merged**